.github/workflows/run-tests.yml

```diff
@@ -18,15 +19,34 @@ jobs:
           java-version: 11
           distribution: temurin
           cache: maven
+      - name: Check formatting in core
+        run: mvn speedy-spotless:check --batch-mode --no-transfer-progress -q -f ./dhis-2/pom.xml
+      - name: Check formatting in web
+        run: mvn speedy-spotless:check --batch-mode --no-transfer-progress -q -f ./dhis-2/dhis-web/pom.xml
+      - name: Check maven dependencies
+        run: mvn dependency:analyze --threads 2C --batch-mode -no-transfer-progress --update-snapshots -f ./dhis-2/pom.xml
 
+  unit-test:
+    runs-on: ubuntu-latest
+    needs:
+      - run-checks
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: temurin
+          cache: maven
       - name: Test core
-        run: mvn clean install --threads 1C --batch-mode -no-transfer-progress --update-snapshots -f ./dhis-2/pom.xml -pl -dhis-web-embedded-jetty
-
+        run: mvn clean install --threads 2C --batch-mode -no-transfer-progress --update-snapshots -f ./dhis-2/pom.xml -pl -dhis-web-embedded-jetty
```

Comment:
Since the `mvn clean install` steps are using the default profile, they will run the `analyze-only` checks. You configured it in the default profile and it binds to the verify phase by [default](https://maven.apache.org/plugins/maven-dependency-plugin/analyze-only-mojo.html). Not sure how much that time that takes. We could run `mvn package` [which does everything up to verify](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#Lifecycle_Reference). Are there any other things running in verify, that we actually want to run and would otherwise lose?